connectors/sources/github.py

```diff
@@ -0,0 +1,252 @@
+
+        Args:
+            request (str): API for the request
+
+        Returns:
+            list: Response of the request
+        """
+        return await self._get_session.getitem(url=request)
+
+    @retryable(
+        retries=RETRIES,
+        interval=RETRY_INTERVAL,
+        strategy=RetryStrategy.EXPONENTIAL_BACKOFF,
+    )
+    async def execute_request(self, method, url):
+        """Execute request using _request method of GitHubAPI.
+
+        Args:
+            request (str): API for the request
+
+        Returns:
+            list: Response of the request
+        """
+        status_code, response_headers, data = await self._get_session._request(
+            method="get", url=USER_API, headers=self.headers
+        )
+        if status_code == 200:
+            return response_headers
+        data = data.decode("utf-8")
+        message = json.loads(data)
+        raise Exception(
+            f"Something went wrong while executing request Exception: {message['message']}"
+        )
+
+    @retryable(
+        retries=RETRIES,
+        interval=RETRY_INTERVAL,
+        strategy=RetryStrategy.EXPONENTIAL_BACKOFF,
+    )
+    async def execute_getiter(self, request):
+        """Execute request using getiter method of GitHubAPI.
+
+        Args:
+            request (str): API for the request
+
+        Yields:
+            dict: Response of the request
+        """
+        async for response in self._get_session.getiter(url=request):
+            yield response
+
+    async def filter_repos(self):
+        try:
+            logger.debug("Filtering repositories")
+            public_repos, configured_repos, users_repos = [], [], []
+            for repo in self.repos:
+                if "/" in repo:
+                    public_repos.append(repo)
+                else:
+                    configured_repos.append(repo)
+
+            async for response in self.execute_getiter(request=ENDPOINTS["ALL_REPOS"]):
+                users_repos.append(response["name"])
+            invalid_repo = list(set(configured_repos) - set(users_repos))
+
+            for repo in public_repos:
+                try:
+                    await self.execute_getitem(
+                        request=ENDPOINTS["REPO"].format(repo_name=repo)
+                    )
+                except BadRequest:
+                    invalid_repo.append(repo)
+            return invalid_repo
+        except Exception as exception:
+            logger.exception(
+                f"Error while filtering repositories. Exception: {exception}."
+            )
+            raise
+
+    @cached_property
+    def _get_session(self):
+        self.session = aiohttp.ClientSession(headers=self.headers)
+        return GitHubAPI(
+            self.session,
+            requester="",
+            oauth_token=self.github_token,
+        )
+
+    async def ping(self):
+        await self.execute_getitem(request=ENDPOINTS["USER"])
+
+    async def close_session(self):
```

Comment:
```suggestion
    async def close(self):
```

nit: I think we can call this method `close` as if we're looking strictly at it, it does not only close one session, but two and it also cancels possible pending sleeps.